rfcs/convergence/input-native-element-props.md

```diff
@@ -0,0 +1,192 @@
+  <input type="checkbox" />
+  <label {...slotProps.label} />
+</div>
+```
+Or an `<Input/>` is roughly like this:
+```tsx
+<div>
+  {/* optional bookend before slot */}
+  <div> {/* wrapper visually styled as an input (maybe, TBD) */}
+    {/* optional start slot */}
+    <input type="text" .../>
+    {/* optional end slot */}
+  </div>
+  {/* optional bookend after slot */}
+</div>
+```
+
+Another unique thing about inputs is that to the degree possible, we'd like them to work nicely with 3rd-party form libraries, which may have APIs that expect to be able to pass native props to the component and have them applied to the actual `<input>`. (needs more research)
+
+### [Formik](https://github.com/formium/formik)
+
+### [react-hook-form](https://github.com/react-hook-form/react-hook-form)
+
+Note, for this one they'd need to make a wrapper for our component because the library expects to be able to *set* `.value` on the input, which we probably don't intend to support. (Someone brought this up for v8 [here](https://github.com/microsoft/fluentui/issues/18126).)
+
+### [React Final Form](https://github.com/final-form/react-final-form)
+
+
+
+<!--
+Why are we making this change? What problem are we solving? What do we expect to gain from this?
+
+This section is important as the motivation or problem statement is indepenent from the proposed change. Even if this RFC is not accepted this Motivation can be used for alternative solutions.
+
+In the end, please make sure to present a neutral Problem statement, rather than one that motivates a particular solution
+-->
+
+## Detailed Design or Proposal
+
+<!-- This is the bulk of the RFC. Explain the proposal or design in enough detail for the inteded audience to understand. -->
+
+We don't have one suggested/preferred solution yet, because all of them have some significant problems.
+
+The examples below assume you're rendering a Checkbox and this is roughly your desired HTML output (it would also include a generated ID to associate the label and input):
+```html
+<div>
+  <input type="checkbox" name="foo" checked />
+  <label>sample</label>
+</div>
+```
+
+### Option 1: All native props always applied to root (current behavior)
+
+Given this:
+```tsx
+<Checkbox name="foo" checked>sample</Checkbox>
+```
+
+You get roughly this HTML, which is not useful:
+```html
+<div name="foo" checked>
+  <input type="checkbox" />
+  <label>sample</label>
+</div>
+```
+
+So to specify a value, you'd have to use slot props:
+```tsx
+<Checkbox input={{ name: 'foo', checked: true }}>sample</Checkbox>
+```
+
+#### Pros
+
+- Complete consistency between all components
+
+#### Cons
+
+- Very counterintuitive API
+    - We could use typings to ensure that doing e.g. `<Checkbox checked/>` caused a TS error, but it still seems very unnatural
+- Might not work as easily with 3rd-party form libraries? (needs more research)
+
+### Option 2: Most native props applied to root, selected props applied to "actual" element
```

Comment:
To a certain extent this would happen anyway right ? since the component itself would most likely be exposing some kind of `value` prop for controlling